platform/ios/src/MLNMapView.mm

```diff
@@ -4241,15 +4249,6 @@ - (void)_flyToCamera:(MLNMapCamera *)camera edgePadding:(UIEdgeInsets)insets wit
             dispatch_async(dispatch_get_main_queue(), pendingCompletion);
         };
     }
-    
-    if ([self.camera isEqualToMapCamera:camera] && UIEdgeInsetsEqualToEdgeInsets(_contentInset, insets))
-    {
-        if (pendingCompletion)
-        {
-            [self animateWithDelay:duration animations:pendingCompletion];
-        }
-        return;
-    }
```

Comment:
This was removed otherwise we wouldn't move the camera if someone calls the new `flyToCamera` method with `UIEdgeInsetsZero` after calling `flyToCamera` with the same camera, but with not zero `UIEdgeInsets`.

This is exactly the case that the example shows.

Reply:
How does this affect current users of this method?

Reply:
Here are 3 proposals how we can proceed with this:

#### Don't do an early return aka keep the lines deleted
This would be consistent with the android sdk since they are not handling this case. I think its totally fine for the consumers of this library to make sure that there is no ongoing camera animation when they start a new one.
We can keep and open issue and come up with a solution before releasing the next major version bump.

#### Add the old check again and document the behavior. 
We can add the behavior into the docs and write `This method will not do anything if the camera didn't change but only the padding changed`. This would be my personal preference from all the options since we have to backport the changes to 5.13 in order to fix another bug in the flutter sdk.

#### Hold the latest `mbgl::CameraOptions` as a member var and don't do anything if the options are the same
This might be the "best" solution but I don't think this should be part of the java/iOS sdks. This can be implemented in C and all platforms can benefit from it. If we do this later on we will fix android automatically as well.

We would like to have this method asap deployed since this is currently blocking us. We can continue our discussion in the slack channel which might be a faster way of communicating

Reply:
Option 1 would be more consistent to android. Of course its debatable if its correct!

Reply:
Option 1 (eliminating the special case) seems risky to me because it’s inconsistent with past behavior and also inconsistent with the other camera-related methods.

I think it’s possible to fix the problem you ran into without introducing other weird behavior. Ignoring the implementation for a moment, the developer would want the method to be a no-op if the camera effectively keeps the viewport in the same position.

Obviously you’d have to compare the current camera to the passed-in camera, but I don’t think you’d want to factor in the content inset, because any change to the content inset has already kicked off a camera change automatically. So the only other comparison you’d make is whether the passed-in edge insets match the excess padding in `mbgl::CameraOptions::padding` caused by mapbox/mapbox-gl-native#15233. You can get the current camera options by calling `getCameraOptions()` on `mbglMap`. Then it’s a matter of subtracting `_contentInset` from `padding` to find that excess padding left over from the previous camera change.

If this doesn’t work, I think your next best bet would be option 1. As long as the passed-in `duration` is negative, this code would ensure that the map only jumps to the target camera (imperceptibly to the user rather than bouncing up and down:

https://github.com/maplibre/maplibre-native/blob/3c57d37ef26ca84d11e376a08a22f00e60c237ce/src/mbgl/map/transform.cpp#L276-L291

Reply:
I reverted the change.